cmake/Utils.cmake

```
@@ -59,17 +59,24 @@ MACRO(ADD_TARGET_PROPERTY TARGET PROPERTY VALUE)
     ENDIF()
 ENDMACRO(ADD_TARGET_PROPERTY)
 
-MACRO(GET_APP_VERSION APP_DIR VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
-    FILE(READ "${APP_DIR}/version" VERSION_RAW)
-    STRING(REGEX MATCHALL "[0-9]" VERSION_PARTS ${VERSION_RAW})
-    LIST(GET VERSION_PARTS 0 ${VERSION_MAJOR})
-    LIST(GET VERSION_PARTS 1 ${VERSION_MINOR})
-    LIST(GET VERSION_PARTS 2 ${VERSION_MAINTENANCE})
+MACRO(GET_APP_VERSION GIT_DESCRIBE VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
+    STRING(REGEX MATCH "v([0-9]+)[.]([0-9]+)[.]([0-9]+)" GIT_DESCRIBE_MATCH ${GIT_DESCRIBE})
+
+    IF(GIT_DESCRIBE_MATCH)
+        SET(${VERSION_MAJOR} ${CMAKE_MATCH_1})
+        SET(${VERSION_MINOR} ${CMAKE_MATCH_2})
+        SET(${VERSION_MAINTENANCE} ${CMAKE_MATCH_3})
+    ELSE()
+        MESSAGE(WARNING "Couldn't parse major/minor/maintenance versions from git describe output '${GIT_DESCRIBE}'")
```

Please let it fail in this case.